lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -389,7 +392,11 @@ def shuffle_positions_on_intermediate_items(old_position, new_position, avoid_id
 
         def insert_at_position(position, raise_exception_if_save_fails=false)
           raise ArgumentError.new("position cannot be lower than top") if position < acts_as_list_top
-          return set_list_position(position, raise_exception_if_save_fails) if new_record?
+          if invalid?
```

Comment:
Can you explain this one? :)

Reply:
Previously, if you have a uniqueness constraint on `position_column` but not a validation, you would get a postgres error when using `insert_at_position` with a pre-existing position (and the transaction would be aborted). However, if you had a uniqueness validation then it'd just raise RecordNotValid which you could catch and continue from.

I've changed `set_list_position` to use raw SQL to update the position, which means the uniqueness validation is no longer checked. In order to maintain the previous behavior (raising an exception instead of rolling back the transaction), I `save!` the record if it's `invalid?`.

Reply:
Flow control via exception handling isn't my preferred style, but if you prefer it I'm happy to change it.

Reply:
This method is becoming too long in general and needs to be split out. I'm having a real hard time to grok it in its current state. Also, if the method names are descriptive enough we should be able to do away with the large amount of comments in here which should further increase readability :) Sorry to be a pain.